src/main/kotlin/org/rust/lang/core/completion/RsImplTraitMemberCompletionProvider.kt

```diff
@@ -52,36 +66,58 @@ object RsImplTraitMemberCompletionProvider : RsCompletionProvider() {
             parentItems.removeIf { it.javaClass == item.javaClass && it.name == item.name }
         }
 
+        // Look for fn/const/type before the caret and filter out the corresponding items
+        val keyword = getPreviousKeyword(element)?.let {
+            val abstractable = KEYWORD_TO_ABSTRACTABLE[it.elementType] ?: return@let null
+            it to abstractable
+        }
+        if (keyword != null) {
+            parentItems.removeIf { it.javaClass != keyword.second }
```

Comment:
Why this line is needed? All tests passes if I remove it.

Reply:
This was actually a bug in tests. Without this line, here:
```rust
trait Trait {
    fn foo(&self);
    type BAR;
}

impl Trait for () {
    fn /*caret*/
}
```
it would offer `type BAR`.

If we are e.g. after `fn`, we want to offer only functions from the parent trait.